### PR TITLE
[Docs] Notes about adding library via single package.json

### PR DESCRIPTION
**Pull request by maruilian11**
for #7793

### CODE REVIEW

**Comment by PeterDaveHello** [comment]
We are going to drop the WIP version of doc, try another one.

Please also add `[no-ci] in commit message` for document PR.

**Comment by maruilian11** [comment]
@PeterDaveHello I turn to add it to README.md.

File: README.md [hunk 1]
```diff
@@ -30,6 +30,70 @@ Please raise a new pull request for new library additions and existing library u
 
 ## Adding a new or updating an existing library
 
+### Using your browser
```
Comment (PeterDaveHello) [inline README.md]: This is not about "browser", the browser has nothing to do with this topic.

File: README.md [hunk 2]
```diff
@@ -30,6 +30,70 @@ Please raise a new pull request for new library additions and existing library u
...
+
+There is a simple way to add a library via single package.json only.
+You can follow these steps:
+
+1.Create a new folder named **libname** under `/ajax/libs`, then create a package.json under `/libname`.
```
Comment (PeterDaveHello) [inline README.md]: You should leave a space after `1.`

File: README.md [hunk 3]
```diff
@@ -30,6 +30,70 @@ Please raise a new pull request for new library additions and existing library u
...
+    {
+      "basePath": "dist",
+      "files": [
+        "frzr*.js"
+      ]
+    }
+  ]
+}
+```
+you can change to use git auto-update cc [autoupdate.md](https://github.com/cdnjs/cdnjs/blob/master/documents/autoupdate.md#enabling-npmrecommended-or-git-auto-update)
```
Comment (PeterDaveHello) [inline README.md]: cc ?

File: README.md [hunk 4]
```diff
@@ -30,6 +30,70 @@ Please raise a new pull request for new library additions and existing library u
...
+
+> Note:
+
+> Please use SPDX in license cc [license-list.json](https://github.com/cdnjs/cdnjs/blob/master/tools/license-list.json)
```
Comment (PeterDaveHello) [inline README.md]: cc again? and why to quote note?

**Comment by maruilian11** [comment]
@PeterDaveHello I changed the title, leave a space after `1.` and use quote to replace `cc`.

**Comment by PeterDaveHello** [comment]
The steps are wrong.

**Comment by maruilian11** [comment]
@PeterDaveHello I add the Fork step and turn `commit  new file` to `propose new file`.

**Comment by PeterDaveHello** [comment]
no mention about `version`?

**Comment by maruilian11** [comment]
@PeterDaveHello mention it in note of package.json.

File: README.md [hunk 5]
```diff
@@ -30,6 +30,75 @@ Please raise a new pull request for new library additions and existing library u
...
+
+    > Please point to minified file in filename
+
+4. Propose new file.
+
+5. Create a pull request.
+
+6. We will check the PR and merge it.
+
+###Others
```
Comment (PeterDaveHello) [inline README.md]: Why call "Others" ?

Comment (maruilian11) [inline README.md]: sorry, I mean "Besides".
is "Besides" ok?

Comment (PeterDaveHello) [inline README.md]: I don't think so.

Comment (maruilian11) [inline README.md]: how about we just put the paragraph under `6. We will check the PR and merge it.`, not adding a title for it.

File: README.md [hunk 6]
```diff
@@ -30,6 +30,75 @@ Please raise a new pull request for new library additions and existing library u
...
+
+    > Please use SPDX in license, [license-list.json](https://github.com/cdnjs/cdnjs/blob/master/tools/license-list.json)
+
+    > Please point to minified file in filename
+
+4. Propose new file.
+
+5. Create a pull request.
+
+6. We will check the PR and merge it.
```
Comment (PeterDaveHello) [inline README.md]: 1. That's not the work for who sent the PR.
2. We will only merge PR when it archived the requirement.

File: README.md [hunk 7]
```diff
@@ -30,6 +30,75 @@ Please raise a new pull request for new library additions and existing library u
...
+        }
+      ]
+    }
+    ```
+    you can change to use git auto-update, [autoupdate.md](https://github.com/cdnjs/cdnjs/blob/master/documents/autoupdate.md#enabling-npmrecommended-or-git-auto-update)
+
+    > Note:
+
+    > Don't add version in package.json if you don't add it with real file
+
```
Comment (PeterDaveHello) [inline README.md]: Do we need this additional new line here?

File: README.md (hunk 5 above)
Comment (PeterDaveHello) [inline README.md]: So why we have title `Add a new library via a single package.json` ? You should make them the same level.

File: README.md [hunk 8]
```diff
@@ -30,6 +30,71 @@ Please raise a new pull request for new library additions and existing library u
...
+        }
+      ]
+    }
+    ```
+    you can change to use git auto-update, [autoupdate.md](https://github.com/cdnjs/cdnjs/blob/master/documents/autoupdate.md#enabling-npmrecommended-or-git-auto-update)
+
+    > Note:
+    
+    > Don't add version in package.json if you don't add it with real file
+
```
Comment (maruilian11) [inline README.md]: @PeterDaveHello I need to add these lines to wrap texts, or do you think we should change to another markdown syntax?

Comment (PeterDaveHello) [inline README.md]: No we don't need these in quote.

File: README.md [hunk 9]
```diff
@@ -30,6 +30,68 @@ Please raise a new pull request for new library additions and existing library u
 
 ## Adding a new or updating an existing library
 
+There is a simple way to add a library via single package.json only.
+You can follow these steps:
```
Comment (PeterDaveHello) [inline README.md]: If you want a new line here, you need a blank line here.

File: README.md [hunk 10]
```diff
@@ -30,6 +30,68 @@ Please raise a new pull request for new library additions and existing library u
...
+        "frzr",
+        "tiny",
+        "view",
+        "library",
+        "turboboosted",
+        "framework",
+        "javascript",
+        "frontend",
+        "client-side",
+        "dom"
```
Comment (PeterDaveHello) [inline README.md]: We don't need so many keywords in example.

File: README.md [hunk 11]
```diff
@@ -30,6 +30,68 @@ Please raise a new pull request for new library additions and existing library u
...
+    }
+    ```
+    you can change to use git auto-update, [autoupdate.md](https://github.com/cdnjs/cdnjs/blob/master/documents/autoupdate.md#enabling-npmrecommended-or-git-auto-update)
+
+    > Note:
+    Don't add version in package.json if you don't add it with real file.
+    Please use SPDX in license, [license-list.json](https://github.com/cdnjs/cdnjs/blob/master/tools/license-list.json).
+    Please point to minified file in filename.
+
+4. Propose new file.
```
Comment (maruilian11) [inline README.md]: ok, I will change to `Submit the package.json`.

**Comment by maruilian11** [comment]
@Piicksarn @cdnjs/intern2 I update the document, could you please help me check this PR? Thanks!

File: README.md [hunk 12]
```diff
@@ -30,6 +30,62 @@ Please raise a new pull request for new library additions and existing library u
...
+          ]
+        }
+      ]
+    }
+    ```
+    you can change to use git auto-update, [autoupdate.md](https://github.com/cdnjs/cdnjs/blob/master/documents/autoupdate.md#enabling-npmrecommended-or-git-auto-update)
+
+    > Note:
+    Don't add version in package.json if you don't add it with real file.
+    Please use SPDX in license, see [SPDX License List](https://spdx.org/licenses/) and [license-list.json](https://github.com/cdnjs/cdnjs/blob/master/tools/license-list.json).
```
Comment (PeterDaveHello) [inline README.md]: Just give SPDX the hyperlink, no need to add additional "SPDX License List"

File: README.md [hunk 13]
```diff
@@ -30,6 +30,62 @@ Please raise a new pull request for new library additions and existing library u
...
+            "frzr*.js"
+          ]
+        }
+      ]
+    }
+    ```
+    you can change to use git auto-update, [autoupdate.md](https://github.com/cdnjs/cdnjs/blob/master/documents/autoupdate.md#enabling-npmrecommended-or-git-auto-update)
+
+    > Note:
+    Don't add version in package.json if you don't add it with real file.
```
Comment (PeterDaveHello) [inline README.md]: `Don't` -> `Please don't`

Comment (PeterDaveHello) [inline README.md]: `if you don't add it with real file` -> `if you don't want to commit with its assets`

File: README.md [hunk 14]
```diff
@@ -30,6 +30,62 @@ Please raise a new pull request for new library additions and existing library u
...
+          "files": [
+            "frzr*.js"
+          ]
+        }
+      ]
+    }
+    ```
+    you can change to use git auto-update, [autoupdate.md](https://github.com/cdnjs/cdnjs/blob/master/documents/autoupdate.md#enabling-npmrecommended-or-git-auto-update)
+
+    > Note:
```
Comment (PeterDaveHello) [inline README.md]: Don't quote the note here, just make them listed.

**Comment by maruilian11** [comment]
@Piicksarn @cdnjs/intern2 I updated the commit, could you please help me check this PR? Thanks!

**Comment by pvnr0082t** [comment]
LGTM

File: README.md [hunk 15]
```diff
@@ -30,6 +30,61 @@ Please raise a new pull request for new library additions and existing library u
...
 
+There is a simple way to add a library via single package.json only.
+
+You can follow these steps:
+
+1. Fork cdnjs repository to your GitHub account.
+
+2. Create a new folder named **libname** under `/ajax/libs`, then create a package.json under `/libname`.
+
+    > A tip: you can type `/libname/package.json` after cdnjs/ajax/libs/ to create them if you use web interface
```
Comment (PeterDaveHello) [inline README.md]: Don't quote this, why you quote your own words even in the first time it appears?

File: README.md [hunk 16]
```diff
@@ -30,6 +30,61 @@ Please raise a new pull request for new library additions and existing library u
...
+
+    > Note: the libname must match the name in package.json
```
Comment (PeterDaveHello) [inline README.md]: be the same as the "name" in package.json

**Comment by maruilian11** [comment]
@PeterDaveHello sorry, I don't really understand, do you mean this two lines

>cdnjs relies on user-submitted pull requests and automatic updating via npm or git repository to populate and update libraries.
To add a new library, or update an existing library outside of npm/git, start by forking the cdnjs repo to your own GitHub account.

should be after this line
## Adding a new or updating an existing library
and be before this line

There is a simple way to add a library via single package.json only.

**Comment by PeterDaveHello** [comment]
@maruilian11 Of course ...

**Comment by maruilian11** [comment]
@PeterDaveHello I updated the changes.

File: README.md [hunk 17]
```diff
@@ -34,6 +34,59 @@ cdnjs relies on user-submitted pull requests and automatic updating via `npm` or
...
 
+There is a simple way to add a library via single package.json only.
+
+You can follow these steps:
+
+1. Fork cdnjs repository to your GitHub account.
+
+2. Create a new folder named **libname** under `/ajax/libs`, then create a package.json under `/libname`.
+
+    > Note: the libname must be the same as the "name" in package.json
```
Comment (PeterDaveHello) [inline README.md]: WHY QUOTE AGAIN?

File: README.md [hunk 18]
```diff
@@ -34,6 +34,59 @@ cdnjs relies on user-submitted pull requests and automatic updating via `npm` or
...
 To add a new library, or update an existing library outside of `npm`/`git`, start by [forking the cdnjs repo](https://github.com/cdnjs/cdnjs/fork) to your own GitHub account.
 
+There is a simple way to add a library via single package.json only.
+
+You can follow these steps:
+
+1. Fork cdnjs repository to your GitHub account.
+
+2. Create a new folder named **libname** under `/ajax/libs`, then create a package.json under `/libname`.
+
```
Comment (PeterDaveHello) [inline README.md]: Why leading slash for `/libname`?

**Comment by maruilian11** [comment]
@PeterDaveHello I removed the slash and quote.

**Comment by PeterDaveHello** [comment]
@maruilian11 Please don't use meaningless branch name like patch-1 in future.

**Comment by maruilian11** [comment]
@PeterDaveHello got it.

**Comment by PeterDaveHello** [comment]
Please rebase with the latest master branch and update commit message to the title, remove `[no-ci]` since it's not supported by drone ci.

**Comment by PeterDaveHello** [comment]
@Piicksarn @pvnr0082t please do a double review to make sure this is good to you to understand.

**Comment by pvnr0082t** [comment]
LGTM

**Comment by maruilian11** [comment]
@pvnr0082t I just rebase the commit with the latest master branch and change the commit msg to fit title, could you please check this PR again? thank you!

**Comment by pvnr0082t** [comment]
LGTM

**Comment by maruilian11** [comment]
@Piicksarn could you help me check this PR? Thanks~

**Comment by maruilian11** [comment]
ping @Piicksarn

**Comment by PeterDaveHello** [comment]
Any updates here? Thanks!

**Comment by maruilian11** [comment]
https://github.com/cdnjs/cdnjs/blob/master/CONTRIBUTING.md#e-adding-a-new-library-by-a-single-packagejson
This PR is no longer needed.

**Pull request closed without merging**